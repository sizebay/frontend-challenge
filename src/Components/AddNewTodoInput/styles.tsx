import styled from "styled-components";
import { theme } from "../../GlobaStyles/theme";

export const InputContainer = styled.form`
  width: 100%;
  display: flex;
  height: 48px;

    input{ 
        width: 100%;
        border: 1px solid ${theme.colors.gray200} ;
        border-radius: 4px 0 0 4px;
        padding-left: 20px;

        ::placeholder{
            color: ${theme.colors.gray500};
            opacity: 0.5;
        }
    }
    button{
        display: grid;
        place-content: center;
        place-items: center;
        border: none;
        width: 44px;
        background-color: ${theme.colors.cyan500};
        border-radius: 0 4px 4px 0;
        transition: opacity 0.2s ease-in-out;
    }
`;
