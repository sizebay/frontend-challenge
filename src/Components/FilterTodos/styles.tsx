import styled from 'styled-components';
import { theme } from '../../GlobaStyles/theme';

interface LabelTypes {
    selected: boolean
}

export const FilterTodosContainer = styled.div`
  display: flex;
  gap: 3.3rem;
  justify-content: space-between;
  margin-bottom: 1.5rem;

  @media(max-width: 800px) {
    flex-direction: column-reverse;
    gap: .9rem;
    margin-bottom: 1rem;
  }
`;

export const FilterButtonsContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 1rem;


`

export const TagButton = styled.button<LabelTypes>`
    position: relative;
    border: 1px solid ${theme.colors.gray200};
    padding: .5rem 1rem;
    border-radius: 300px;
    transition: all 0.2s ease-in-out;

    ${props => props.selected && `
        border: 1px solid ${theme.colors.cyan500};
        color ${theme.colors.cyan500};
        background-color: ${theme.colors.gray100};
        ::before{
            content: "âœ“";

    `}

`
export const FilterInputContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    border: 1px solid ${theme.colors.gray200};
    width: 100%;
`

export const FilterInput = styled.input`
    width: 100%;
    height: 40px;
    outline: none;
    padding-left: 1rem;
    border: none;
`

