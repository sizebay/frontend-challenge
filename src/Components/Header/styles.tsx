import styled from 'styled-components';
import { theme } from '../../GlobaStyles/theme';

export const HeaderContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  margin-bottom: 24px;
`;

export const DateHolder = styled.div`
  display: grid;
  justify-content: center;
  grid-template-areas:
    "DateNumber DateMonth"
    "DateNumber DateYear";
`;

export const DateNumber = styled.p`
  grid-area: DateNumber;
  font-size: ${theme.font.size[60]};
  font-weight: ${theme.font.weight.medium};
`;

export const DateMonth = styled.span`
  grid-area: DateMonth;
  align-self: end;
  font-size: ${theme.font.size[24]};
  font-weight: ${theme.font.weight.regular};
  width: 3ch;
  white-space: nowrap;
  overflow: hidden;
`;

export const DateYear = styled.span`
  grid-area: DateYear;
  font-size: ${theme.font.size[24]};
  font-weight: ${theme.font.weight.light};
`;

export const WeekDay = styled.p`
  font-size: ${theme.font.size[24]};
  font-weight: ${theme.font.weight.regular};
`;
