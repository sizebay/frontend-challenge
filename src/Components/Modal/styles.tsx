import styled from 'styled-components';
import { theme } from '../../GlobaStyles/theme';

export const ModalContainer = styled.main`
  width: 100vw;
  max-width: 800px;
  height: 650px;
  background-color: ${theme.colors.white};
  border-radius: 4px;
  padding: 54px 60px;

  
  @media(max-width: 800px) {
    height: 90vh;
    max-width: 90vw;
  }
`;
