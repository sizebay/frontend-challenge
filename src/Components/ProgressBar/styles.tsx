import styled from 'styled-components';
import { theme } from '../../GlobaStyles/theme';

interface ProgressType {
    percentage: number;
}

export const ProgressBarContainer = styled.div`
  height: 24px;
  width: 100%;
  background-color: ${theme.colors.gray100};
  border-radius: 4px;
  margin-bottom: 24px;
  overflow: hidden;
`;

export const Progress = styled.div<ProgressType>`
  width: ${props => props.percentage}%;
  background-color: ${theme.colors.green500};
  height: 100%;

  transition: width 0.2s ease-in-out;
`;

