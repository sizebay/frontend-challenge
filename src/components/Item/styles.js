import styled from 'styled-components';

import minusImg from '../../assets/minus.svg';
import checkImg from '../../assets/check.svg';

export const ItemContainer = styled.form`
  position: relative;
  display: flex;
  border: 1px solid #DBDBDB;
  border-radius: 4px;
  background: #F4F4F4;
  color: #848484;
  margin-bottom: 1rem;
  margin-right: 10px;
  height: 3.5rem;

  input {
    width: 100%;
    background: none;
    color: #848484;
    font-size: 1rem;
    border: none;
    padding-left: 1rem;

    &:focus {
      outline: none;
      border: none;
    }
  }

  &:focus-within > button {
    display: inline-block;
  }

  &:focus-within > input {
    background: #fff 0% 0% no-repeat padding-box;
    border-radius: 0.4rem 0 0 0.4rem;
    position: relative;
  }
`;

export const ItemContainerNoEdit = styled.form`
  display: flex;
  border: 1px solid #DBDBDB;
  border-radius: 4px;
  background: #F4F4F4;
  color: #848484;
  margin-bottom: 1rem;
  margin-right: 10px;
  height: 3.5rem;

  input {
    width: 100%;
    background: none;
    color: #848484;
    font-size: 1rem;
    border: none;
    padding-left: 1rem;
    cursor: not-allowed;

    &:focus {
      outline: none;
      border: none;
    }
  }

  &:focus-within > button {
    display: inline-block;
  }

  &:focus-within > input {
    border-radius: 0.4rem 0 0 0.4rem;
    position: relative;
  }
`;

export const RemoveButton = styled.button`
  display: ${(props) => (props.show ? 'block' : 'none')};
  width: 3rem;

  cursor: pointer;
  border: transparent;
  outline: transparent;
  background: #e34f4f 0% 0% no-repeat padding-box;

  &::before {
    content: "";
    background-image: url(${minusImg});
    background-color: rgb(255, 255, 255);
    background-repeat: no-repeat;
    background-position: center;
    display: inline-block;
    width: 1.34rem;
    height: 1.34rem;
    border-radius: 50%;
  }
`;

export const DoneButton = styled.button`
  display: ${(props) => (props.show ? 'block' : 'none')};
  flex-direction: center;
  align-items: center;
  width: 3rem;
  cursor: pointer;
  border: transparent;
  outline: transparent;
  background: #5de290 0% 0% no-repeat padding-box;

  &::before {
    content: "";
    background-image: url(${checkImg});
    background-color: rgb(255, 255, 255);
    background-repeat: no-repeat;
    background-position: center;
    display: inline-block;
    width: 1.34rem;
    height: 1.34rem;
    border-radius: 50%;
  }
`;

export const ItemTooltip = styled.div`
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 5.34vh;
  width: 4.47vw;
  background: #848484;
  color: #fff;
  top: 80%;
  left: 40%;
  border-radius: 4px;
  z-index: 1;
  padding-left: 0.5rem;
  padding-right: 0.5rem;

  -webkit-touch-callout: none;
  -webkit-user-select: none;
  -khtml-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;

  &::after {
    content: "";
    position: absolute;
    top: -18%;
    left: 50%;
    margin-left: -5px;
    border-width: 5px;
    border-style: solid;
    border-color: #848484 transparent transparent transparent;
    transform: rotate(180deg);
  }

  @media (max-width: 768px) {
    display: none;
  }
`;
