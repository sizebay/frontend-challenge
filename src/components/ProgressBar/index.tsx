import React from 'react';
import { StyledProgressBar, ProgressFill } from './style';

export interface ProgressBarProps {
  completedItems: number;
    totalTasks: number;
  }

function ProgressBar({ completedItems, totalTasks }: ProgressBarProps) {
  const progressBarWidth = (completedItems / totalTasks) * 100 || 0;

  return ( 
    <StyledProgressBar>
        <ProgressFill width={progressBarWidth} completed={completedItems === totalTasks} />
    </StyledProgressBar>
  );
};

export default ProgressBar;