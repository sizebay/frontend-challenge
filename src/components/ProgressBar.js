import styles from '../css/ProgressBar.module.css'

function ProgressBar({percentage}){
    const widthStyle = {
        width: `${percentage}%`
    }

    return(
        <div className={styles.containerStyles}>
            <div style={widthStyle} className={styles.fillerStyles}>
            </div>
        </div>
    )
}

export default ProgressBar;