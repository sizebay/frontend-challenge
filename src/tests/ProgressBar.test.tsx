import React from 'react';
import { render } from '@testing-library/react';
import ProgressBar from '../components/ProgressBar/index';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import '@testing-library/jest-dom';

describe('ProgressBar component', () => {
  it('renders correctly with default props', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <ProgressBar completedItems={0} totalTasks={10} />
      </ThemeProvider>
    );

    const progressFill = getByTestId('progress-fill');

    expect(progressFill).toHaveStyle('width: 0%');
    expect(document.body).toMatchSnapshot();
  });

  it('renders correctly with completed items', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <ProgressBar completedItems={5} totalTasks={10} />
      </ThemeProvider>
    );

    const progressFill = getByTestId('progress-fill');

    expect(progressFill).toHaveStyle('width: 50%');
  });

  it('renders as completed when all items are completed', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <ProgressBar completedItems={10} totalTasks={10} />
      </ThemeProvider>
    );

    const progressFill = getByTestId('progress-fill');

    expect(progressFill).toHaveStyle('width: 100%');
  });
});
