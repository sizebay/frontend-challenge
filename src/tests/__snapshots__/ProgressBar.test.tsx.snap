// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar component renders correctly with default props 1`] = `
<body>
  <div>
    <div
      class="sc-aXZVg gHGjTk"
      data-testid="progress-bar"
    >
      <div
        class="sc-gEvEer cHAHfT"
        data-testid="progress-fill"
        width="0"
      />
    </div>
  </div>
</body>
`;
